src/Catalog/Persistence/Storage.cs

```diff
@@ -191,8 +205,23 @@ protected void TraceMethod(string method, Uri resourceUri)
         {
             if (Verbose)
             {
-                Trace.WriteLine(String.Format("{0} {1}", method, resourceUri));
+                //The Uri depends on the storage implementation.
+                Uri storageUri = GetUri(GetName(resourceUri));
+                Trace.WriteLine(String.Format("{0} {1}", method, storageUri));
             }
         }
+
+        private string TraceException(string method, Uri resourceUri, Exception exception)
+        {
+            string message = $"{method} EXCEPTION: {GetUri(GetName(resourceUri))} {exception.Message}";
+            Trace.WriteLine(message);
+            return message;
+        }
+
+        private void TraceExecutionTime(string method, Uri resourceUri, long executionTimeInMilliseconds)
+        {
+            string message = $"ExecutionTime_Method:{method}_Uri:{GetUri(GetName(resourceUri))}_Milliseconds:{executionTimeInMilliseconds}";
```

Comment:
Why is this encoded so?

Reply:
The goal is to have a string format to be parsed in AI. The goal is to have a string format to be parsed in AI.

Reply:
Spoke offline. @cristinamanum is going to check is JSON serialization is easier from the AI side.

Reply:
Updated.